aws/s3_test.go

```diff
@@ -458,3 +462,57 @@ func TestNukeS3Bucket(t *testing.T) {
 		})
 	}
 }
+
+// TestFilterS3BucketArgs represents arguments for TestFilterS3Bucket_Config
+type TestFilterS3BucketArgs struct {
+	configFilePath string
+	matches        []string
+}
+
+func bucketNamesForConfigTests() []string {
+	return []string{
+		"alb-alb-123456-access-logs-" + S3TestGenBucketName(),
+		"alb-alb-234567-access-logs-" + S3TestGenBucketName(),
+		"tonico-prod-alb-access-logs-" + S3TestGenBucketName(),
+		"prod-alb-public-access-logs-" + S3TestGenBucketName(),
+		"stage-alb-internal-access-logs-" + S3TestGenBucketName(),
+		"stage-alb-public-access-logs-" + S3TestGenBucketName(),
+	}
+}
+
+// TestFilterS3Bucket_Config tests listing only S3 buckets that match config file
+func TestFilterS3Bucket_Config(t *testing.T) {
```

Comment:
I originally wrote several integration tests against a set of buckets, testing different sets of filters (include, exclude, both together), but it will be a nightmare to maintain those tests. We know the filtering is parsing correctly based on the unit tests (config_test.go), so it makes sense to limit the integration test to the "both together" (include and exclude) case, which Is what I've done here.

Would like some feedback on the effectiveness of this kind of test.

Reply:
I'm going to add the exclude-only and include-only integration tests back in because it will help with any changes in code review. I've concluded that we probably need them for confidence.